File: rxnormalize/src/main/java/io/sweers/rxnormalize/RxNormalize.java
package io.sweers.rxnormalize;

import java.util.concurrent.TimeUnit;

import rx.Observable;
import rx.Observable.Transformer;
import rx.schedulers.Schedulers;

public final class RxNormalize {
  private RxNormalize() {
    throw new AssertionError("No instances.");
  }

  public static <T> Transformer<T, T> transform(final long time, final TimeUnit unit) {
    return new Transformer<T, T>() {
      @Override
      public Observable<T> call(Observable<T> source) {
        return source
            .lift(new OperatorNormalize<T>(time, unit, Schedulers.computation()));
      }
    };
  }
}

File: rxnormalize-kotlin/src/main/kotlin/io/sweers/rxnormalize/RxNormalize.kt
package io.sweers.rxnormalize

import rx.Observable
import rx.Scheduler
import rx.schedulers.Schedulers
import java.util.concurrent.TimeUnit

/**
 * Adds a normalize() operator to `Observable`
 */
public inline fun <T> Observable<T>.normalize(window: Long, unit: TimeUnit): Observable<T> = normalize(window, unit, Schedulers.computation())

/**
 * Adds a normalize() operator to `Observable`
 */
public inline fun <T> Observable<T>.normalize(window: Long, unit: TimeUnit, scheduler: Scheduler): Observable<T> = lift(OperatorNormalize<T>(window, unit, scheduler))
